Feature/#6 harvest learning data

Qbicz opened the pull request:
Prepared a module which may be used in scripts for calculating Hu moments in learning phase.
- we now compute Hu moments of a filled shape, not a contour

Qbicz committed:
Hu module: add prototype

Qbicz committed:
create modules dir with print_vector

Qbicz committed:
svm train: Makefile linking modules src directory

Qbicz committed:
Hu module: fill contours

Qbicz committed:
Hu moments: compute shape descriptors for filled contour

Qbicz committed:
Hu: fix error in cv::moments() by providing 1-channel image

Qbicz committed:
Hu: update help

palucki commented on PC/hu_moments_gather/hu_moments_gather.cpp:
```diff
@@ -0,0 +1,106 @@
+#include <vector>
+
+#include <opencv2/highgui/highgui.hpp>
+#include "opencv2/imgproc/imgproc.hpp"
```
We should be consistent, and use <> for files included files, as long as it's not a file located in the repo.

palucki commented on PC/hu_moments_gather/hu_moments_gather.cpp:
```diff
@@ -0,0 +1,106 @@
+
+#include "print_vector.h"
+
+#define SHOW_IMAGES     1
```
maybe CLI parameter instead of #define ? (Needs no recompilation)

Qbicz replied:
Do you know a nice way to have named CLI arguments to C++ programs?

palucki commented on PC/svm/svm_train_and_save/Makefile:
```diff
@@ -1,19 +1,24 @@
+
 CXX=g++
-CXXFLAGS= `pkg-config --libs opencv` -std=c++11
+CXXFLAGS= `pkg-config --libs opencv` -std=c++11 -I$(ROOT_DIR)/modules
```
+1

palucki commented on PC/svm/svm_train_and_save/svm_train_and_save.cpp:
```diff
@@ -23,14 +34,27 @@ int main()
     for (auto element : training_data_json)
     {
+        // Read training data
+        std::vector<float> training_element = svm_build_single_element(element);
```
Do we want to do it manually every time?
Why not prepare_data_mat(training_data_json); ?

Qbicz replied:
You are right, will prepare

palucki reviewed (commented):
Needs a little bit of rework, but looks promising!

Qbicz committed:
Hu: fixed includes and added cmd-line argument requested in review

Qbicz committed:
Hu: reorganize bash<->C++ program interface

Now we invoke program only once with list of images and get JSON file with array of objects,
each representing Hu moments and a learning label. Previously we had to invoke the program from bash several times,
each program call apending and object to JSON array in file which was was unfeasible.

palucki commented on PC/hu_moments_gather/hu_moments_gather.cpp:
```diff
@@ -20,26 +23,57 @@ int canny_threshold;
     }
 
-    if(argc == 3)
+    if(argc >= 3)
```
Shouldn't be:
`if(argc == 3)` ???
Having 4 arguments we will not initialize
`show_images_cli_arg`, will we?

palucki reviewed (changes requested):
All we need to fix the parameters (As mentioned in comment) and can merge it

Qbicz committed:
PC/camera: program to capture images from a Logitech camera

Qbicz committed:
hu_moments_gather: fix paths to images. Now able to obtain JSON

Qbicz committed:
hu_moments: instructions for machine learning

Qbicz committed:
hu_moments: fixed CLI argument handling

Qbicz committed:
hu_moments: clean up global declarations

Qbicz committed:
hu_moments: save vector to JSON as string
